interfaces/api/service/IAuthorizationChecker.h

```diff
@@ -0,0 +1,88 @@
+/**
+ * @copyright Copyright (c) 2021-2023 B-com http://www.b-com.com/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IAUTHORIZATIONCHECKER_H
+#define IAUTHORIZATIONCHECKER_H
+
+#include <xpcf/api/IComponentIntrospect.h>
+#include <xpcf/api/IComponentManager.h>
+#include <xpcf/core/helpers.h>
+#include "core/Messages.h"
+#include "core/SolARFrameworkDefinitions.h"
+
+namespace SolAR {
+namespace api {
+namespace service {
+
+/**
+ * @class IAuthorizationChecker
+ * @brief <B>Checks if a client providing a token is authorized to use a service.</B>
+ * <TT>UUID: 02031edc-2388-470e-84e5-4d997fd1f093</TT>
+ */
+
+class XPCF_IGNORE SOLARFRAMEWORK_API IAuthorizationChecker :
+    virtual public org::bcom::xpcf::IComponentIntrospect {
+public:
+    /// @brief IFrontEnd default constructor
+    IAuthorizationChecker() = default;
+
+    /// @brief IFrontEnd default destructor
+    virtual ~IAuthorizationChecker() = default;
+
+    /// @brief Verifiy if the token is valid.
+    /// @param[in] accessToken the token to verify
+    /// @return
+    /// * FrameworkReturnCode::_SUCCESS if the the token is valid
+    /// * FrameworkReturnCode::_AUTHENT_INVALID_TOKEN if the authentication token is invalid
+    /// * else FrameworkReturnCode::_ERROR_
+    virtual FrameworkReturnCode verify(const std::string & accessToken) = 0;
```

Comment:
nit: check and verify basically means the same thing, maybe be more specific?
* `isValid()`, `checkValidity()`, `checkAccessTokenValidity()`,...?
* `isAuthorized()`, `checkAuthorization()`,... ?

Reply:
Done, renamed isTokenValid()